Signal/src/util/ThreadUtil.m

```diff
@@ -239,18 +239,40 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
             // Due to disappearing messages or manual deletion,
             // firstUnseenInteractionTimestampParameter may refer to an obsolete
             // interaction in which case we want to discard it.
-            TSInteraction *_Nullable lastInteraction =
-                [[transaction ext:TSMessageDatabaseViewExtensionName] lastObjectInGroup:thread.uniqueId];
```

Comment:
This approach was too simplistic - it didn't handle the case where there were info or error messages.